data/blood/glucose/glucose.go

```diff
@@ -60,3 +61,49 @@ func NormalizeValueForUnits(value *float64, units *string) *float64 {
 	}
 	return value
 }
+
+const (
+	// MgdL_Per_MmolL renames MmolLToMgdLConversionFactor in an alternate way.
+	MgdL_Per_MmolL = MmolLToMgdLConversionFactor
+	// MmolL_Per_MgdL inverts the MmolL_Per_MgdL conversion factor.
+	MmolL_Per_MgdL = 1 / MgdL_Per_MmolL
+)
+
+// Convert between common blood glucose measurement units.
+//
+// Results are rounded to five decimal places. However, if fromUnits and
+// toUnits are the same, the value is returned unchanged. If either fromUnits
+// or toUnits aren't handled, Convert panics.
+//
+// Convert is modeled to return results as similar as possible to
+// NormalizeValueForUnits, but without the hassle of pointer values, and with
+// the ability to convert between arbitrary units (as supported).
+func Convert(value float64, fromUnits, toUnits string) float64 {
+	from, to := normalizeUnitsCase(fromUnits), normalizeUnitsCase(toUnits)
+	if from == to {
+		return value
+	}
+	if from == MgdL && to == MmolL {
+		return Round5(value * MmolL_Per_MgdL)
+	}
+	if from == MmolL && to == MgdL {
+		return Round5(value * MgdL_Per_MmolL)
```

Comment:
I'm not sure this is going to do what you want. You are rounding a mg/dL value to 5 when only mmol/L should be rounded to 5. This is why I think we should stick with original values and use explicit limits for both mg/dL and mmol/L rather than doing conversion first to apply the limit to a common unit.

See problem here. https://go.dev/play/p/qRTDdyktgq8

Reply:
Can you go deeper into what's wrong here? The fact that you don't get exactly the same value back is an expected behavior of unit conversions with limited precision. Is there some other problem with the logic?

If you take it one step further, and go back to mmol/L again, you'll see you get the same answer as the first time:

https://go.dev/play/p/rqGmk8Ajw25

So it does enter a steady state. The conversions aren't 1:1 by their nature, so you can never have perfect bi-directional conversion. That's why the choice of precision is important, it's saying that if there are any differences in conversion, at this precision, they're small enough that they don't matter.

Or am I misunderstanding something else about your comment?

Reply:
As a followup, in terms of significant digits (6), if we round the result to 6 significant digits, it becomes 140.000, and that works out.

Reply:
Either way, given our discussion with @ps2 today, this code will be removed, so no worries.